packages/core/src/shared/globalState.ts

```diff
@@ -8,15 +8,31 @@ import { getLogger } from './logger/logger'
 import * as redshift from '../awsService/redshift/models/models'
 import { TypeConstructor, cast } from './utilities/typeConstructors'
 
+type samInitStateKey =
+    | 'ACTIVATION_TEMPLATE_PATH_KEY'
+    | 'ACTIVATION_LAUNCH_PATH_KEY'
+    | 'SAM_INIT_RUNTIME_KEY'
+    | 'SAM_INIT_IMAGE_BOOLEAN_KEY'
+    | 'SAM_INIT_ARCH_KEY'
+
 type globalKey =
+    | samInitStateKey
     | 'aws.downloadPath'
     | 'aws.lastTouchedS3Folder'
     | 'aws.lastUploadedToS3Folder'
     | 'aws.redshift.connections'
     | 'aws.toolkit.amazonq.dismissed'
     | 'aws.toolkit.amazonqInstall.dismissed'
+    | 'aws.toolkit.separationPromptDismissed'
+    | 'aws.toolkit.separationPromptCommand'
+    | 'aws.amazonq.codewhisperer.newCustomizations'
     // Deprecated/legacy names. New keys should start with "aws.".
```

Comment:
In fact these keys are not deprecated, they are still in use.

Reply:
The comment is just meant to express the fact that the naming style should not be copied.